src/pages/Result.tsx

```diff
@@ -57,7 +99,9 @@ const Result: React.FC<Props> = ({ resultId }) => {
                                 {songData.lyrics.warped.map(({ text }) => <LyricsLine>{text}</LyricsLine> )}
                             </LyricsWrapper>
                         </SongWrapper>
-                        <GalleryWrapper/>
+                        <GalleryWrapper>
+                            {renderMemeCreationTools(memeCreationStage, windowSelection)}
```

Comment:
Just a nice early note, this seems like an absolute prime usecase for Reach Router nested-routes (https://reach.tech/router/example/nested-routes). At the very least for handling a final state (to make it easily shareable via URL).

If we don't want to do that I think it might make more sense to just turn this into a new component `MemeTools` which internally handles the current `stage` in state. Which I think makes this slightly nicer from a component API/composability perspective. The page doesn't *really* care about what `stage` creation is in, it just cares if it's complete, or if it's cancelled etc.